5.0 prerelease

GoldenSoju opened the pull request:
Solves [#298 (Change of actual RecurrenceRule Class)](https://github.com/builttoroam/device_calendar/issues/298).

Breaking Change:
- Replace old Recurrence Rule implementation with [rrule package](https://pub.dev/packages/rrule).

Other Changes:
- Use of Flutter Lints
- Add **event status**
- Several dependency updates
- Sample App updates/improvements
- Other fixes/improvements

Needed before final release:
- Readme/Changelog update

Should we do a prerelease (for one week maybe)?

thomassth committed:
fix for ownerAccount null

thomassth committed:
more calendar data displayed

thomassth committed:
example app dark mode

GoldenSoju committed:
Implementing Rrule package (#403)

Co-authored-by: GoldenSoju <pol82fa91!>
Co-authored-by: thomassth <thomassth@users.noreply.github.com>

thomassth committed:
activate flutter lint

thomassth committed:
more linting fixes

thomassth committed:
Merge pull request #443 from SungHyun22/ownerAccount-null

Fixes owner account null (#427)

thomassth committed:
Update issue templates

thomassth commented:
I was thinking 4.3 but that works too

I think we can do 2 weeks prerelease just to be safe

Also, I was thinking about adding some compatibility to the old recurrence rule, to make the switchover easier

thomassth committed:
Merge pull request #444 from builttoroam/thomassth-patch-2

Update issue templates

thomassth committed:
packages cleanup

Commit:
Fix create/update events in IOS

Fixed fullDay with DateRange events creation (not only one fullday events possible to create)
Timezone should be set only if event is not full day

thomassth committed:
Merge pull request #449 from thomassth/reduce-import

packages cleanup

Commit:
Fixed formatting

Commit:
Fixed file formatting with spaces

GoldenSoju committed:
iOS: Fix error when adding attendees due to UUID is nil.

IVLIVS-III committed:
Merge pull request #453 from GoldenSoju/ios_attendees_fix

IVLIVS-III committed:
Merge pull request #450 from OleksandraFedotova/develop

Fixed All day events start time may be incorrect (iOS only) #337

tjarvstrand committed:
Remove faulty requestPermissions implementation in favor of working one

IVLIVS-III committed:
Merge pull request #465 from tjarvstrand/fix-request-permissions

Remove faulty requestPermissions implementation in favor of working one

sgowda97 committed:
parseRecurrenceRuleString() when else fix.

IVLIVS-III committed:
Fixed failing test (dart analyze).

IVLIVS-III committed:
Bumped version and updated changelog.

VladyslavBilomeria committed:
Fix retrieving events for more than 4 years range

VladyslavBilomeria committed:
Simplify the range size calculation algorithm

VladyslavBilomeria committed:
Remove logs-related code

IVLIVS-III committed:
Merge pull request #486 from VladyslavBilomeria/fix/ios-retrieve-events

Fix retrieving events for more than 4 years range

IVLIVS-III committed:
Merge branch 'release' into develop

IVLIVS-III committed:
Updated example project for iOS.

IVLIVS-III committed:
Fixed deprecation issues.

IVLIVS-III committed:
Merge pull request #487 from builttoroam/align-develop-with-master

Align develop with master

ebsangam committed:
add proguard rules

IVLIVS-III committed:
Merge pull request #489 from ebsangam/develop

junqi committed:
Request FullAccess on iOS17+ (#497)

* fix: request FullAccess on iOS17 or later

* update iOS integration description

* fix: iOS build test & Android build test

Commit:
included namespace

IVLIVS-III committed:
Merge pull request #508 from ash80/develop

Included namespace for Android Gradle Plugin Version 8.0+ and Gradle Version 8.0

thomassth committed:
pipeline update

IVLIVS-III committed:
Merge pull request #513 from builttoroam/workflow-update

Commit:
building fixes

Commit:
added event color for android

Commit:
added removing of event color

Commit:
added updating of calendar color for android and ios

Commit:
fixes for iOS updateCalendarcolor

Commit:
small fix

Commit:
added calendar color  file

Commit:
made colorKey accessible in event

Commit:
set eventcolor, loaded eventColorKey for event

Commit:
mvoed color picker file

Commit:
removed old color picker

Commit:
exposed color  and colorKey

Commit:
fixed color picker import

Commit:
fixes for setting event color

Commit:
clean up

Commit:
added serilization test for eventColor

Commit:
tiny formatting adjustment

Commit:
set example kotlin version to 1.8..22

Commit:
flutter_native_timezone made release builds fail, migrated to flutter_timezone

Commit:
increased  rrule version from 0.2.10 to 0.2.15

IVLIVS-III committed:
Merge pull request #546 from chris-wolf/develop

Added event color (Android) + Updating of calendar color (Android/iOS)

Jonny5-5 committed:
Fix isAllDay null error on iOS

When an Event object is created in flutter with null allDay value the value will default to false on iOS like it does on Android.

IVLIVS-III committed:
Merge pull request #583 from Jonny5-5/allday-null-error
